### PR TITLE
Documentation around temporal versioning

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Add documentation about versioning in temporal workflow.

### CODE REVIEW

**Review by jrhizor — APPROVED** [review]

**Comment by benmoriceau** [comment]
@jrhizor I forget to ask you about the placement of this file. Does it sounds right to you?

**Review by cgardens — APPROVED** [review]
great. i think you can just put this in `airbyte-workers/readme.md`. it will be more discoverable than if you nest it in the package.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Temporal Development
+
+## Versioning
+
+Temporal is maintaining an internal history of the activity it runs. This history is based on a specific order. If we restart a temporal workflow with
+a new implementation that have a different order, the workflow will be stuck and will need manual action to be properly restarted. Temporal provides
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md:6]:
```suggestion
a new implementation that has a different order, the workflow will be stuck and will need manual action to be properly restarted. Temporal provides
```
```suggestion
a new implementation that have a different order, the workflow will be stuck and will need manual action to be properly restarted. Temporal provides
```

Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md:6]:
```suggestion
a new implementation that has a different order, the workflow will be stuck and will need manual action to be properly restarted. Temporal provides
```

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+and API to be able to manage those changes smoothly. However, temporal is very permissive in the way to generate those version. Airbyte will follow
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md:7]:
```suggestion
an API to be able to manage those changes smoothly. However, temporal is very permissive with version rules. Airbyte will follow
```

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+the following rules:
+
+- There will be one global version per workflow, meaning that we will use a single tag per workflow.
+- All the following code modification will need to bump the version number, it won't be limited to a release of a new airbyte version
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md:11]:
```suggestion
- All the following code modifications will need to bump the version number, it won't be limited to a release of a new airbyte version
```

**Review by davinchia — APPROVED** [review]
I appreciate the documentation! thank you!

Agree with Charles on moving this to the top level read me.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md (hunk 1 above)
Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md:6]: Done

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md (hunk 2 above)
Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/README.md:7]: Done

**Pull request merged**